src/validators/AllowedSessionsValidator.sol

```diff
@@ -0,0 +1,127 @@
+  /// that define the behavior and constraints of a session.
+  function setSessionActionsAllowed(
+    bytes32 sessionActionsHash,
+    bool allowed
+  ) external virtual onlyRole(SESSION_REGISTRY_MANAGER_ROLE) {
+    isSessionSpecAllowed[sessionActionsHash] = allowed;
+    emit SessionActionsAllowed(sessionActionsHash, allowed);
+  }
+
+  /// @notice Get the hash of session actions from a session specification.
+  /// @param sessionSpec The session specification.
+  /// @return The hash of the session actions.
+  /// @dev The session actions hash is derived from the session's fee limits, call policies, and transfer policies.
+  function getSessionActionsHash(SessionLib.SessionSpec memory sessionSpec) public view virtual returns (bytes32) {
+    return keccak256(abi.encode(sessionSpec.feeLimit, sessionSpec.callPolicies, sessionSpec.transferPolicies));
+  }
+
+  /// @notice Create a new session for an account.
+  /// @param sessionSpec The session specification to create a session with.
+  /// @dev A session is a temporary authorization for an account to perform specific actions, defined by the session specification.
+  function createSession(SessionLib.SessionSpec memory sessionSpec) public virtual override {
+    bytes32 sessionActionsHash = getSessionActionsHash(sessionSpec);
+    if (!isSessionSpecAllowed[sessionActionsHash]) {
+      revert SessionActionsNotAllowed(sessionActionsHash);
+    }
+    super.createSession(sessionSpec);
+  }
+
+  /// @inheritdoc SessionKeyValidator
+  function supportsInterface(
+    bytes4 interfaceId
+  ) public pure override(SessionKeyValidator, AccessControl) returns (bool) {
+    return
+      interfaceId == type(IERC165).interfaceId ||
+      interfaceId == type(IModuleValidator).interfaceId ||
+      interfaceId == type(IModule).interfaceId ||
+      interfaceId == type(IAccessControl).interfaceId;
+  }
+
+  /// @notice Validate a session transaction for an account.
+  /// @param signedHash The hash of the transaction.
+  /// @param transaction The transaction to validate.
+  /// @return true if the transaction is valid.
+  /// @dev Session spec and period IDs must be provided as validator data.
+  function validateTransaction(
+    bytes32 signedHash,
+    Transaction calldata transaction
+  ) external virtual override returns (bool) {
+    (bytes memory transactionSignature, address _validator, bytes memory validatorData) = SignatureDecoder
+      .decodeSignature(transaction.signature);
+    (SessionLib.SessionSpec memory spec, uint64[] memory periodIds) = abi.decode(
+      validatorData, // this is passed by the signature builder
+      (SessionLib.SessionSpec, uint64[])
+    );
+    if (spec.signer == address(0)) {
+      revert Errors.SESSION_ZERO_SIGNER();
+    }
+    bytes32 sessionActionsHash = getSessionActionsHash(spec);
+    if (!isSessionSpecAllowed[sessionActionsHash]) {
+      revert SessionActionsNotAllowed(sessionActionsHash);
+    }
```

Comment:
We can check for this right away after decoding the session spec from the signature, and avoid code duplication by calling `super.validateTransaction`. Unless we're after gas savings by avoiding the double decoding, but I'm not sure how significant they are.

Reply:
You're right, but the thing is `external`, so ofc I could do: `AllowedSessionsValidator(address(this)).validateSignature` but it seems a bit odd here, since it's neither diamond proxy nor some other one =)

Reply:
Ofc, I'd prefer it also that way =)